app/src/main/java/com/mk/tasky/core/presentation/TaskyTextField.kt

```diff
@@ -0,0 +1,157 @@
+            errorCursorColor = Red,
+            errorIndicatorColor = Red,
+            errorLabelColor = Red,
+            textColor = DarkGray,
+            placeholderColor = Gray,
+            leadingIconColor = Gray,
+            backgroundColor = Light2,
+            focusedIndicatorColor = LightBlue,
+            unfocusedIndicatorColor = Color.Transparent,
+            trailingIconColor = if (isValid) Green else trailingIconColor ?: Gray,
+            cursorColor = DarkGray
+        ),
+        placeholder = {
+            Text(text = placeholder)
+        },
+        isError = showError,
+        maxLines = maxLines,
+        singleLine = maxLines == 1,
+        trailingIcon = {
+            TrailingIcon(
+                isPassword = isPassword,
+                trailingIcon = trailingIcon,
+                onIconClick = onIconClick,
+                isValid = isValid,
+                onPasswordClick = {
+                    isPasswordHidden = !isPasswordHidden
+                },
+                isPasswordHidden = isPasswordHidden
+            )
+        },
+        shape = RoundedCornerShape(10.dp),
+        visualTransformation = if (isPassword && isPasswordHidden) PasswordVisualTransformation() else VisualTransformation.None
+    )
+}
+
+@Composable
+fun TaskyEmailTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String = "",
+    showError: Boolean = false,
+    isValid: Boolean = false,
+    modifier: Modifier = Modifier
+) {
+    TaskyTextField(
+        value = value,
+        onValueChange = onValueChange,
+        placeholder = placeholder,
+        showError = showError,
+        modifier = modifier,
+        keyboardType = KeyboardType.Email,
+        isValid = isValid
+    )
+}
+
+@Composable
+fun TaskyPasswordTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String = "",
+    showError: Boolean = false,
+    isValid: Boolean = false,
+    modifier: Modifier = Modifier
+) {
+    TaskyTextField(
+        value = value,
+        onValueChange = onValueChange,
+        placeholder = placeholder,
+        showError = showError,
+        modifier = modifier,
+        keyboardType = KeyboardType.Password,
+        isPassword = true,
+        isValid = isValid
+    )
+}
+
+@Composable
+private fun TrailingIcon(
+    isPassword: Boolean = false,
+    isPasswordHidden: Boolean,
+    trailingIcon: ImageVector? = null,
+    isValid: Boolean = false,
+    onPasswordClick: () -> Unit,
+    onIconClick: () -> Unit = {}
+) {
+    if (trailingIcon != null) {
+        IconButton(onClick = onIconClick) {
+            Icon(imageVector = trailingIcon, contentDescription = "icon")
+        }
+    } else if (isPassword) {
+        IconButton(onClick = onPasswordClick) {
+            val icon =
+                if (isPasswordHidden) Icons.Default.VisibilityOff else Icons.Default.Visibility
+            Icon(imageVector = icon, contentDescription = "icon")
```

Comment:
The purpose of a content description is to provide accessibility support (for example for blind people, so the device can tell them what an image is for). "icon" isn't really descriptive, I'd rather use something that tells them what this icon button is used for, for example "Toggle password visibility". 
And in a perfect world, this could also support localization by using string resources 😆

Reply:
Changed the text

In the future I'll extract it to string resources or make it possible to receive via parameter